### PR TITLE
migrate all entity pages to composability api

### PR DESCRIPTION
**Pull request by Rugvip**
Final and biggest piece of the composability migration. 🎉  This won't build or pass tests atm and I haven't tried it out manually either, it's just the code changes the we intend to have working in the future. Before that we also need a few external updates, e.g. https://github.com/RoadieHQ/backstage-plugin-travis-ci/issues/52. We ofc also need to port the app template and probably a couple of pieces in docs as well.

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4822-iqqtm3vxydn1mjce2sutxraeuz0qnpkv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/604159eaf285af04bfa0bbc7

File: packages/app/src/components/catalog/EntityPage.tsx [hunk 1]
```diff
@@ -14,490 +14,437 @@
...
 
-import {
-  ApiEntity,
-  DomainEntity,
-  Entity,
-  GroupEntity,
-  SystemEntity,
-  UserEntity,
-} from '@backstage/catalog-model';
+import React from 'react';
```
Comment (Rugvip) [inline packages/app/src/components/catalog/EntityPage.tsx:17]: Just making sure this file is expanded in the diff view :>

**Comment by Fox32** [comment]
> This won't build or pass tests atm and I haven't tried it out manually either, it's just the code changes the we intend to have working in the future.

I think this declaimer is quite clear 😆 All of our dependencies are migrated, so I thought I give it a try and ended up with:

```
index.esm.js?ca4d:2185 Uncaught Error: Visited element WithStyles(ForwardRef(Grid)) twice
    at eval (index.esm.js?ca4d:2185)
    at forEachSingleChild (react.development.js?e1d5:1118)
    at traverseAllChildrenImpl (react.development.js?e1d5:1007)
    at traverseAllChildren (react.development.js?e1d5:1092)
    at Object.forEachChildren [as forEach] (react.development.js?e1d5:1140)
    at traverseElementTree (index.esm.js?ca4d:2178)
    at eval (index.esm.js?ca4d:2604)
    at mountMemo (react-dom.development.js?03bb:15443)
    at Object.useMemo (react-dom.development.js?03bb:15739)
    at useMemo (react.development.js?e1d5:1521)
eval @ index.esm.js?ca4d:2185
forEachSingleChild @ react.development.js?e1d5:1118
traverseAllChildrenImpl @ react.development.js?e1d5:1007
traverseAllChildren @ react.development.js?e1d5:1092
forEachChildren @ react.development.js?e1d5:1140
traverseElementTree @ index.esm.js?ca4d:2178
eval @ index.esm.js?ca4d:2604
mountMemo @ react-dom.development.js?03bb:15443
useMemo @ react-dom.development.js?03bb:15739
useMemo @ react.development.js?e1d5:1521
wrappedHook @ react-hot-loader.development.js?0e8f:2798
Provider @ index.esm.js?ca4d:2603
renderWithHooks @ react-dom.development.js?03bb:14804
mountIndeterminateComponent @ react-dom.development.js?03bb:17483
beginWork @ react-dom.development.js?03bb:18597
callCallback @ react-dom.development.js?03bb:189
invokeGuardedCallbackDev @ react-dom.development.js?03bb:238
invokeGuardedCallback @ react-dom.development.js?03bb:293
beginWork$1 @ react-dom.development.js?03bb:23204
performUnitOfWork @ react-dom.development.js?03bb:22155
workLoopSync @ react-dom.development.js?03bb:22131
performSyncWorkOnRoot @ react-dom.development.js?03bb:21757
scheduleUpdateOnFiber @ react-dom.development.js?03bb:21189
updateContainer @ react-dom.development.js?03bb:24374
eval @ react-dom.development.js?03bb:24759
unbatchedUpdates @ react-dom.development.js?03bb:21904
legacyRenderSubtreeIntoContainer @ react-dom.development.js?03bb:24758
render @ react-dom.development.js?03bb:24841
eval @ index.tsx?63ed:8
Promise.then (async)
eval @ index.tsx?63ed:7
eval @ index.tsx:29
./src/index.tsx @ main.chunk.js:693
__webpack_require__ @ runtime.js:854
fn @ runtime.js:151
1 @ main.chunk.js:743
__webpack_require__ @ runtime.js:854
checkDeferredModules @ runtime.js:46
webpackJsonpCallback @ runtime.js:33
(anonymous) @ main.chunk.js:1
react-dom.development.js?03bb:19528 The above error occurred in the <Provider> component:
    in Provider (at App.tsx:117)
    in App (at src/index.tsx:8)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
logCapturedError @ react-dom.development.js?03bb:19528
logError @ react-dom.development.js?03bb:19565
update.callback @ react-dom.development.js?03bb:20709
callCallback @ react-dom.development.js?03bb:12491
commitUpdateQueue @ react-dom.development.js?03bb:12512
commitLifeCycles @ react-dom.development.js?03bb:19884
commitLayoutEffects @ react-dom.development.js?03bb:22804
callCallback @ react-dom.development.js?03bb:189
invokeGuardedCallbackDev @ react-dom.development.js?03bb:238
invokeGuardedCallback @ react-dom.development.js?03bb:293
commitRootImpl @ react-dom.development.js?03bb:22542
unstable_runWithPriority @ scheduler.development.js?8189:653
runWithPriority$1 @ react-dom.development.js?03bb:11040
commitRoot @ react-dom.development.js?03bb:22382
finishSyncRender @ react-dom.development.js?03bb:21808
performSyncWorkOnRoot @ react-dom.development.js?03bb:21794
scheduleUpdateOnFiber @ react-dom.development.js?03bb:21189
updateContainer @ react-dom.development.js?03bb:24374
eval @ react-dom.development.js?03bb:24759
unbatchedUpdates @ react-dom.development.js?03bb:21904
legacyRenderSubtreeIntoContainer @ react-dom.development.js?03bb:24758
render @ react-dom.development.js?03bb:24841
eval @ index.tsx?63ed:8
Promise.then (async)
eval @ index.tsx?63ed:7
eval @ index.tsx:29
./src/index.tsx @ main.chunk.js:693
__webpack_require__ @ runtime.js:854
fn @ runtime.js:151
1 @ main.chunk.js:743
__webpack_require__ @ runtime.js:854
checkDeferredModules @ runtime.js:46
webpackJsonpCallback @ runtime.js:33
(anonymous) @ main.chunk.js:1
index.esm.js?ca4d:2185 Uncaught (in promise) Error: Visited element WithStyles(ForwardRef(Grid)) twice
    at eval (index.esm.js?ca4d:2185)
    at forEachSingleChild (react.development.js?e1d5:1118)
    at traverseAllChildrenImpl (react.development.js?e1d5:1007)
    at traverseAllChildren (react.development.js?e1d5:1092)
    at Object.forEachChildren [as forEach] (react.development.js?e1d5:1140)
    at traverseElementTree (index.esm.js?ca4d:2178)
    at eval (index.esm.js?ca4d:2604)
    at mountMemo (react-dom.development.js?03bb:15443)
    at Object.useMemo (react-dom.development.js?03bb:15739)
    at useMemo (react.development.js?e1d5:1521)
```

I wonder if that happends if I for example reuse a subtree like `componentOverviewContent` more than once.

**Comment by Rugvip** [comment]
@Fox32 https://github.com/backstage/backstage/pull/5000 :grin:

**Comment by Fox32** [comment]
> @Fox32 #5000 😁

Thanks, I will give it a try now

**Comment by Rugvip** [comment]
@Fox32 I have actually tried this out myself now without the Roadie plugins, and it does render entity pages x)

**Comment by Fox32** [comment]
> @Fox32 I have actually tried this out myself now without the Roadie plugins, and it does render entity pages x)

Works for me too! Just have to check whether my routing problem is also solved.

**Comment by Fox32** [comment]
@Rugvip routing in entity pages works as expected, thank you!

**Comment by Rugvip** [comment]
This one's actually ready for review now 😅

File: packages/app/src/components/catalog/EntityPage.tsx [hunk 2]
```diff
@@ -14,174 +14,96 @@
...
-    case isRollbarAvailable(entity):
-      return <RollbarRouter entity={entity} />;
-    default:
-      return <SentryRouter entity={entity} />;
-  }
-};
+// import {
+//   EntityBuildkiteContent,
+//   isBuildkiteAvailable,
+// } from '@roadiehq/backstage-plugin-buildkite';
```
Comment (Rugvip) [inline packages/app/src/components/catalog/EntityPage.tsx:87]: Commented out the plugins that can't be used yet, and then let's add them back as they become available

Comment (Fox32) [reply packages/app/src/components/catalog/EntityPage.tsx:87]: cc @dtuite did you already see this?

**Review by Fox32 — APPROVED** [review]
From someone that has already done it for his own instance: 👍 looks good and also nice to have the step by step explanation in the changeset.

File: packages/app/src/components/catalog/EntityPage.tsx (hunk 2 above)
Comment (dtuite) [reply packages/app/src/components/catalog/EntityPage.tsx:87]: @Fox32 Nope. Just seeing it now. We will get this fixed ASAP.

Comment (Xantier) [reply packages/app/src/components/catalog/EntityPage.tsx:87]: @Rugvip Can you elaborate what the issue on buildkite plugin is preventing it from being used?

Comment (Rugvip) [reply packages/app/src/components/catalog/EntityPage.tsx:87]: I created some issues around these, https://github.com/RoadieHQ/backstage-plugin-travis-ci/issues/52, and https://github.com/RoadieHQ/backstage-plugin-github-insights/issues/125. One is fixed but the other one is still open, and I believe it applies to all of the plugins.

I think the best way to check what's needed is to try out this branch and uncomment the code, iirc it's what is expected to work. except that the `is*Available` return value should not need to be case to a bool, so fix for that would be nice too :grin:

**Comment by Xantier** [comment]
I pulled this down and was clicking through a bit and noticed that docs within entity pages seems to fail to render subpaths. Eg. going to `http://localhost:3000/catalog/default/component/documented-component/docs/sub-page/` blanks out with `Uncaught Error: Routable extension component was not discovered in the app element tree. ` error. The root of the docs (`http://localhost:3000/catalog/default/component/documented-component/docs/`) seem to render fine.

**Comment by Fox32** [comment]
> I pulled this down and was clicking through a bit and noticed that docs within entity pages seems to fail to render subpaths. Eg. going to `http://localhost:3000/catalog/default/component/documented-component/docs/sub-page/` blanks out with `Uncaught Error: Routable extension component was not discovered in the app element tree. ` error. The root of the docs (`http://localhost:3000/catalog/default/component/documented-component/docs/`) seem to render fine.

We are also using 
```tsx
      <EntityLayout.Route path="/docs" title="Docs">
        <EntityTechdocsContent />
      </EntityLayout.Route>
```
in our app it it works fine 🤔

**Comment by Rugvip** [comment]
@Xantier It was fixed by https://github.com/backstage/backstage/pull/5203, but that's not part of this PR branch afaik

**Comment by Fox32** [comment]
I can confirm, rebasing on master fixes this.

**Review by benjdlambert — APPROVED** [review]

**Comment by Rugvip** [comment]
Shipping so that we can get the create-app changes as part of the release. The plugins are only disabled in the example-app, but let's re-enable them asap.

**Comment by dtuite** [comment]
@Rugvip @Fox32 

Fixes for all the plugins are in progress now, thanks to @Xantier. Apologies for the trouble.

https://github.com/RoadieHQ/backstage-plugin-github-pull-requests/pull/83
https://github.com/RoadieHQ/backstage-plugin-buildkite/pull/41
https://github.com/RoadieHQ/backstage-plugin-github-insights/pull/164/files
https://github.com/RoadieHQ/backstage-plugin-travis-ci/pull/58
https://github.com/RoadieHQ/backstage-plugin-firebase-functions/pull/37
https://github.com/RoadieHQ/backstage-plugin-aws-lambda/pull/44
https://github.com/RoadieHQ/backstage-plugin-argo-cd/pull/56
https://github.com/RoadieHQ/backstage-plugin-security-insights/pull/31

**Pull request merged**